airbyte-commons-server/src/main/java/io/airbyte/commons/server/scheduler/DefaultSynchronousSchedulerClient.java

```diff
@@ -79,14 +83,14 @@ public SynchronousResponse<StandardCheckConnectionOutput> createSourceCheckConne
     final JobCheckConnectionConfig jobCheckConnectionConfig = new JobCheckConnectionConfig()
         .withActorType(ActorType.SOURCE)
         .withActorId(source.getSourceId())
-        .withConnectionConfiguration(sourceConfiguration)
+        .withConnectionConfiguration(configInjector.injectConfig(sourceConfiguration, source.getSourceDefinitionId()))
```

Comment:
When we call `withConnectionConfiguration`, will we always want to have the config injection? If the config is always the same and there will always be injection, should we wrap this in a "ConnectionConfigurationDAO"? Else, we will need to modify everywhere the config is used every time we change something associated to it

Reply:
Something we could do right now that might be a good preparation is to make the `ConfigInjector` also take care of calling the `OAuthConfigSupplier` as they have similar parameters and I think at the moment both of them are called in the same places. OTOH it might be more confusing than helpful to tie these things together right now.